### PR TITLE
--datetime parameter addition

### PR DESCRIPTION
**Pull request by Proph151Music**
The new `--timedate` parameter is a way that users can specify a date and time they would like to download/extract Starchive containers from.  This gives them much more control and can significantly speed up the process. Since they no longer have to download the entire Starchive.

For instance, if the node was offline or out of consensus for 7 days... they could specify that date and Starchiver will look at the hash file to determine which container would be a good starting point to ensure they only download/extract what they need.

Their are multiple ways `--timedate` can be used...

It can be used with a unix style timestamp, also this UTC format "YYYY-MM-DD zHMM" or "YYYY-MM-DD zHMM", it can be this format "YYYY-MM-DD.HH", or if no data is given... it will automatically search for the most recent snapshot timestamp.

Then it will use this data to determine with Starchive container to start at.

### CODE REVIEW

File: starchiver-ext [hunk 1]
```diff
@@ -472,13 +567,101 @@ main_menu() {
...
+            echo "No valid snapshot found within the specified path."
+            exit 1
+        fi
+    fi
+}
+
+function parse_arguments() {
+    while [[ "$#" -gt 0 ]]; do
+        case "$1" in
+            --timedate)
```
Comment (AlexBrandes) [inline starchiver-ext]: You might consider naming this "datetime" instead since that's a commonly used term in programming when combining date and time.

**Review by AlexBrandes — COMMENTED** [review]
This seems fairly dangerous overall, in the sense that it seems likely to create a broken chain. How do you verify that you're pulling the correct next snapshot from the archive compared to what the user has locally?

**Comment by Proph151Music** [comment]
In this version it purposely goes back 1 hour from the timestamp. But it's not perfect since the mtime can be very different on the users server. So... In this case if a user has an issue they can send a specific date and time to go back to.

In the next version I am going to make sure the mtime of all snapshots compiled inside the Starchiver Containers are matching the timestamp of the snapshot on the API. Then I will change this code to always look at the timestamp on the API instead of the mtime of the file itself. 

It will definitely get more testing once it's released though. But I'm pretty confident it will work. :-)

File: starchiver-ext (hunk 1 above)
Comment (Proph151Music) [reply starchiver-ext]: Oh ok.  Good idea! I will make this change today.

Comment (Proph151Music) [reply starchiver-ext]: That actually makes more sense too. Since that's how the data reads.

**Pull request merged**